### PR TITLE
Fix JupyterLab and related refactoring

### PR DESCRIPTION
**Pull request by rtitle**
See: https://github.com/DataBiosphere/leonardo/issues/417

- Fixes the localization endpoint when using JupyterLab
- Fixes Jupyter server logging to `~/jupyter.log`
- Forces python2 runtime when starting the Jupyter to server, to prevent user scripts/extensions from changing the runtime
- Organize scripts inside the docker container into subdirectories, and add a `run-jupyter.sh` start script

Tested manually. Existing tests should cover most of these; one thing we don't have is an auto-test for JupyterLab. I can create a separate ticket to add this.

TODO: update `jupyter-docker/README.md`

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: jupyter-docker/scripts/run-jupyter.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Forces python 2
+JUPYTER_BASE="/usr/bin/python /usr/local/bin/jupyter-notebook"
+JUPYTER_CMD="$JUPYTER_BASE --NotebookApp.nbserver_extensions=\"{'jupyter_localize_extension':True}\" &> ${HOME}/jupyter.log"
```
Comment (rtitle) [inline jupyter-docker/scripts/run-jupyter.sh]: I had to add the `--NotebookApp.nbserver_extensions` to the jupyter command to make it load the `jupyter_localize_extension`. The same thing is also specified in `jupyter_notebook_config.py` but it wasn't getting loaded unless I explicitly added it here.

File: jupyter-docker/scripts/run-jupyter.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+echo $JUPYTER_CMD
+eval $JUPYTER_CMD
```
Comment (vkumra-broad) [inline jupyter-docker/scripts/run-jupyter.sh]: Nit: needs new line

**Comment by vkumra-broad** [comment]
Question: We have two localize/delocalize python files -- `jupyter_localize_extension.py` and `jupyter_delocalize.py`, which one is the one that we are using?

**Comment by rtitle** [comment]
Both - `jupyter_localize_extension.py` adds the `/localize` endpoint for localizing files onto the VM via API. `jupyter_delocalize.py` hooks into Jupyter `ContentsManager` to support delocalizing when the user hits "save" in the Jupyter UI.

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
Test failure should be addressed by https://github.com/DataBiosphere/leonardo/pull/447.

**Pull request merged**